### ISSUE TITLE
"VERSION" and DESIRED VERSION" field is blank while running "kubectl get PravegaCluster"

### ISSUE THREAD

**Issue by Raghuveera-Rao** [pravega/pravega-operator#214]
'kubectl get PravegaCluster' not displaying 'VERSION and 'DESIRED VERSION' with kubectl version 1.13 and pravega-operator:0.3.2

$ kubectl get PravegaCluster
```
NAME      VERSION   DESIRED VERSION   DESIRED MEMBERS   READY MEMBERS   AGE
pravega                               9                 9               3m
```
$kubectl version
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:26:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

```
Deployment detail:
```
bookkeeper:0.5.0-2269.6f8a820
pravega:0.5.0-2269.6f8a820
zookeeper:3.5.4-beta-operator
pravega-operator:0.3.2
zookeeper-operator:0.2.1

```

**Comment by pbelgundi** [comment]
Adding additional printer columns to the CRD can help with displaying information in `kubectl get`:
Please see:
https://v1-12.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns


Kubernetes Objects ( spec and status)
https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status

**Comment by pbelgundi** [comment]
Also see: https://github.com/pravega/pravega-operator/blob/master/deploy/crd.yaml#L12

**Comment by fpj** [comment]
There is something funny here. The upgrade documentation says that in the absence of an upgrade, the cluster status should look like:

```
$ kubectl get PravegaCluster
NAME      VERSION   DESIRED MEMBERS   READY MEMBERS   AGE
example   0.4.0     7                 7               11m
```

https://github.com/pravega/pravega-operator/blob/16b10f759d3fc7000fba28b976d8c3cbd6555e99/doc/upgrade-cluster.md#L66

so no `Desired Version`, which is concept only valid during upgrades. Is an upgrade being attempted here or are we incorrectly printing the report in the absence of an upgrade?

**Comment by Raghuveera-Rao** [comment]
@fpj 
No upgrade tried.. It is fresh installation with 'kubectl version 1.13 '..
 with older version of kubectl(1.10) , I only used to get
```
NAME      AGE
pravega   3m
```

**Comment by fpj** [comment]
Are you installing via helm or are performing a manual deployment?

**Comment by Raghuveera-Rao** [comment]
I am doing manual installation using kubectl commands.

**Comment by SrishT** [comment]
It seems like the 0.3.2 operator is being deployed using a CRD from 0.4.0 operator code. As you can see in https://github.com/pravega/pravega-operator/blob/0.3.2/deploy/crd.yaml the `Version` and `Desired Version` fields were not even displayed in operator 0.3.2

**Comment by pbelgundi** [comment]
Operator 0.3.2 does not have a concept of "Pravega Cluster Version" and there is no 'Version' field in the Pravega Cluster Spec which was added in 0.4.0.

Custer spec in 0.3.2:
https://github.com/pravega/pravega-operator/blob/0.3.2/pkg/apis/pravega/v1alpha1/pravegacluster_types.go#L59

In 0.4.0:
https://github.com/pravega/pravega-operator/blob/master/pkg/apis/pravega/v1alpha1/pravegacluster_types.go#L89

As such the CRD in 0.3.2 also does not provide a 'version' field and so version is not expected to be displayed with 0.3.2 operator.

For 0.3.2, all we could display as version is the pravega image tag... which I don't think is of much value.

**Comment by Raghuveera-Rao** [comment]
$ kubectl get crd pravegaclusters.pravega.pravega.io -o json | grep -i version
```
"apiVersion": "apiextensions.k8s.io/v1beta1",
        "resourceVersion": "6231111",
                "JSONPath": ".status.currentVersion",
                "description": "The current pravega version",
                "name": "Version",
                "JSONPath": ".status.TargetVersion",
                "description": "The desired pravega version",
                "name": "Desired Version",
        "version": "v1alpha1",
        "versions": [
        "storedVersions": [
```
It means I am using 0.4.0 crd . 
Further to that,
$cat ./pkg/apis/pravega/v1alpha1/pravegacluster_types.go | grep Version
```
* Licensed under the Apache License, Version 2.0 (the "License");
        // DefaultPravegaVersion is the default tag used for for the Pravega
        DefaultPravegaVersion = "0.4.0"
        // Version is the expected version of the Pravega cluster.
        Version string `json:"version"`
        if s.Version == "" {
                s.Version = DefaultPravegaVersion
        // Deprecated: Use `spec.Version` instead
```
which further implied **DefaultPravegaVersion = "0.4.0"**

Verified with tag of tags/0.3.2 and also used kubectl 1.13 version

$kubectl get PravegaCluster
```
NAME      DESIRED MEMBERS   READY MEMBERS   AGE
pravega   9                 9               1d

```

**Comment by Raghuveera-Rao** [comment]
The issue was due to taking 0.4.0 crd version and using Pravega-operator 0.3.2.

**Comment by sumit-bm** [comment]
@pbelgundi @SrishT @fpj As per yesterday weekly meeting, I thought the discussion was to include `version` field in 0.3.2? Please correct me.

**Comment by pbelgundi** [comment]
Fix has been applied to "crd.yaml" under deploy folder in 0.3.3, so please use the same to verify this fix.

**Comment by pbelgundi** [comment]
Fixed in 0.3.3
